vm/src/stdlib/itertools.rs

```diff
@@ -269,40 +269,81 @@ mod decl {
         }
     }
 
-    #[pyimpl(with(PyIter))]
+    #[derive(FromArgs)]
+    struct PyRepeatNewArgs {
+        #[pyarg(any)]
+        object: PyObjectRef,
+        #[pyarg(any, optional)]
+        times: OptionalArg<PyIntRef>,
+    }
+
+    #[pyimpl(with(PyIter), flags(BASETYPE))]
     impl PyItertoolsRepeat {
         #[pyslot]
         fn tp_new(
             cls: PyTypeRef,
-            object: PyObjectRef,
-            times: OptionalArg<PyIntRef>,
+            PyRepeatNewArgs { object, times }: PyRepeatNewArgs,
             vm: &VirtualMachine,
         ) -> PyResult<PyRef<Self>> {
-            let times = times
-                .into_option()
-                .map(|int| PyRwLock::new(int.as_bigint().clone()));
-
+            let times = match times.into_option() {
```

Comment:
`times` was changed from a `BigInt` to a `usize` with a max value of `isize::MAX`. This matches current implementation in `CPython` (which uses a `Py_ssize_t`) and does make its execution faster.